### ISSUE TITLE
4.x RFC Honor displayField in templates

### ISSUE THREAD

**Issue by steinkel** [cakephp/bake#597]
Would it make sense to honor displayField in templates generated?

In this case, it would be useful to propagate the displayField configuration to the entity, or adding additional logic to the entity to navigate back to the table and pick the displayField value for output, `$entity->getDisplayFieldValue()`

**Comment by markstory** [comment]
Where would you want to use the displayField where it is not?

**Comment by steinkel** [comment]
default links to associations

**Comment by dereuromark** [comment]
@steinkel Are you making a PR here?

**Comment by steinkel** [comment]
Yes, I'll help with this one, I'll add some logic to navigate back to the table, get the displayField and return the value or default to the 'id' field if not found...

**Comment by dereuromark** [comment]
@steinkel Are you interested in solving this for 4.x?

**Comment by steinkel** [comment]
yes, I'll find some time next week for it. Thanks,

**Comment by steinkel** [comment]
After playing with some alternatives, I see the display field is already used in the bake template here https://github.com/cakephp/bake/blob/master/templates/bake/Template/index.twig#L46 so the problem was updating the value and having to re-run bake to get it applied.

Alternatives to keep this behavior dynamic are not worth it (from my point of view) because:
* Navigating the table from the entity via source would add a dependency to the TableRegistry and logic to check if the alias exists, etc

```php
public function getDisplayFieldValue(): string
    {
        $source = $this->getSource();
        $displayFieldValue = $this->id;
        if ($source) {
            $table = \Cake\ORM\TableRegistry::getTableLocator()->get($source);
            $displayField = $table->getDisplayField();
            if ($this->has($displayField)) {
                return $this->get($displayField);
            }
        }
    }
```

* Propagating the display field to the entity could work, but in this case it would be either duplicated with the table property or we would introduce breaking changes migrating from it to the EntityTrait.

Long story short, I think adding a feature to dynamically update the display field is not worth it as we can re-bake the templates or use crud plugin.

**Comment by ADmad** [comment]
> Long story short, I think adding a feature to dynamically update the display field is not worth it as we can re-bake the templates or use crud plugin.

Righto, hence closing the issue.
